### PR TITLE
Add `getProperty` to `ClassLike`

### PR DESCRIPTION
**Pull request by ruudk**
This makes it easier to get a property by name from a ClassLike.

### CODE REVIEW

File: lib/PhpParser/Node/Stmt/ClassLike.php [hunk 1]
```diff
@@ -53,6 +53,27 @@ public function getProperties() : array {
         return $properties;
     }
 
+    /**
+     * Gets property with the given name defined directly in this class/interface/trait.
+     *
+     * @param string $name Name of the property (compared case-insensitively)
```
Comment (nikic) [inline lib/PhpParser/Node/Stmt/ClassLike.php]: Property names are case-sensitive.

Comment (ruudk) [reply lib/PhpParser/Node/Stmt/ClassLike.php]: Fixed.

**Comment by ruudk** [comment]
Would it be an idea to also add this for `getConstant`?

File: lib/PhpParser/Node/Stmt/ClassLike.php [hunk 2]
```diff
@@ -53,6 +53,26 @@ public function getProperties() : array {
         return $properties;
     }
 
+    /**
+     * Gets property with the given name defined directly in this class/interface/trait.
```
Comment (nikic) [inline lib/PhpParser/Node/Stmt/ClassLike.php]: I think we need to make more clear here that this returns a `Property` which *contains* the desired property, but may also have other properties. So this is sufficient if you want to get the modifiers or property type, but if you need the initializer, you need to fetch the contained `PropertyProperty`.

We should also have a test for the case where the `Property` has multiple properties, and it's sufficient that one of them matches.

Comment (ruudk) [reply lib/PhpParser/Node/Stmt/ClassLike.php]: > I think we need to make more clear here that this returns a Property which contains the desired property, but may also have other properties. So this is sufficient if you want to get the modifiers or property type, but if you need the initializer, you need to fetch the contained PropertyProperty.

What if we also add a `getProperty` method on `Property` that returns the actual PropertyProperty? Together with that, we can improve the description of `ClassLike::getProperty` to mention that method.

> We should also have a test for the case where the Property has multiple properties, and it's sufficient that one of them matches.

Already added, see `fooBarProp` in diff

**Comment by ruudk** [comment]
@nikic Can we merge this PR? Or do you want me to make any adjustments. Let me know, thanks!

**Comment by nikic** [comment]
Sorry for the delay!

**Comment by TomasVotruba** [comment]
Any ETA on release? I'd love to use this in Rector and drop custom method handling it: 

https://github.com/rectorphp/rector/blob/5f4ca314a9c307c28d40551bff1fc1d3ad4d82a7/src/PhpParser/Node/Manipulator/ClassManipulator.php#L213-L231

**Pull request merged**